### PR TITLE
feat: adds user billing functions

### PR DESCRIPTION
**Pull request by Justintime50**
# Description

Adds `PaymentMethod.all()`, `CreditCard.fund()`, and `CreditCard.delete()` functions

# Testing

- [x] Added unit test for retrieve all payment methods
- [x] E2E tested (as much as possible) that deleting a credit card and funding a wallet would at least hit the right endpoints, can't test these functions due to not having a valid CC to use

# Pull Request Type

Please select the option(s) that are relevant to this PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement (fixing a typo, updating readme, renaming a variable name, etc)

### CODE REVIEW

File: easypost/credit_card.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        primary_or_secondary = params.get("primary_or_secondary")
+        payment_method_to_use = payment_method_map.get(primary_or_secondary) if primary_or_secondary else None
+        card_id = payment_methods[payment_method_to_use]["id"] if payment_method_to_use else None
+
+        if payment_method_to_use is None or card_id is None or "card_" not in card_id:
+            raise Error(message="The chosen payment method is not a credit card. Please try again.")
+
+        requestor = Requestor(local_api_key=api_key)
+        url = f"{cls.class_url()}/{card_id}/charges"
+        wrapped_params = {"amount": params.get("amount")}
```
Comment (jchen293) [inline easypost/credit_card.py:36]: I don't see the benefit of using `**params` in this function, can we just have two parameters in this function? 

Also In line 36, what if there is no `amount` in the `**params`? Should we check it before sending it to the API?

**Review by jchen293 — CHANGES_REQUESTED** [review]

File: easypost/credit_card.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        card_id = payment_method_to_use.get("id")
+
+        requestor = Requestor(local_api_key=api_key)
+        url = f"{cls.class_url()}/{card_id}/charges"
+        wrapped_params = {"amount": params.get("amount")}
+        response, api_key = requestor.request(method=RequestMethod.POST, url=url, params=wrapped_params)
+        return convert_to_easypost_object(response=response, api_key=api_key)
+
+    @classmethod
+    def delete(cls, easypost_id: str, api_key: Optional[str] = None):
```
[the remark this answers is not in the document]
Comment (Justintime50) [reply easypost/credit_card.py:41]: I followed the pattern of the `retrieve` function which calls it `easypost_id`: https://github.com/EasyPost/easypost-python/blob/3f792eefd9468efb0ea2773aa72637398686b268/easypost/resource.py#L24, I'm not opposed to being more explicit here though

File: easypost/credit_card.py (hunk 1 above)
Comment (Justintime50) [reply easypost/credit_card.py:36]: Just following the pattern of the lib where we traditionally use `**params` for everything; however, I don't see the signature of this function changing much over time so we could use a more explicit keyword param approach.

Comment (Justintime50) [reply easypost/credit_card.py:36]: As for checking if it's empty etc, I don't feel we should here as there are additional validations needed at the API level anyway. Let's pass it through and make the API catch what's necessary all in one go.

File: easypost/credit_card.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jchen293) [reply easypost/credit_card.py:41]: I prefer to be more explicit, similar to what we have done in the Referral class.

File: easypost/credit_card.py (hunk 1 above)
Comment (jchen293) [reply easypost/credit_card.py]: Got it. In Java, the param only needs `amount` so I wouldn't ask the user to input a Map in the parameter that only has one value.

File: easypost/credit_card.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Justintime50) [reply easypost/credit_card.py]: I've switched it

**Review by jchen293 — APPROVED** [review]

**Review by nwithan8 — APPROVED** [review]

**Pull request merged**